src/zjs_timers.c

```diff
@@ -131,7 +140,7 @@ static jerry_value_t add_timer_helper(const jerry_value_t function_obj,
     zjs_timer_t* handle = add_timer(interval, callback, this, repeat, argv, argc - 2);
     if (handle->callback_id == -1)
         return zjs_error("native_set_interval_handler: timer alloc failed");
-    jerry_set_object_native_handle(timer_obj, (uintptr_t)handle, zjs_timer_free_cb);
```

Comment:
Also I'm no JavaScript expert, but I thought the only reason setInterval should be GC'd is if clearInterval is called or the script ends.  We could pass the zjs_timer_free_cb for setTimeout though.  Would that be more complete in your opinion?